lib/bson/decimal128.js

```diff
@@ -245,9 +242,29 @@ Decimal128.fromString = function(string) {
     throw new Error('' + string + ' not a valid Decimal128 string');
   }
 
-  // Check if we have an illegal exponent format
-  if (stringMatch && stringMatch[4] && stringMatch[2] === undefined) {
-    throw new Error('' + string + ' not a valid Decimal128 string');
+  if (stringMatch) {
+    // full_match = stringMatch[0]
```

Comment:
these are for documentation purposes, not left-over commented out code, correct?

Reply:
yeah, I spent a while figuring out what exactly the regex was giving back, so I figured I'd leave it for future reference.